x-pack/plugins/cloud_security_posture/public/test/fixtures/react_query.ts

```diff
@@ -22,11 +22,15 @@ export const createReactQueryResponse = <TData = unknown, TError = unknown>({
   data = undefined,
 }: CreateReactQueryResponseInput<TData, TError> = {}): Partial<UseQueryResult<TData, TError>> => {
   if (status === 'success') {
-    return { status, data };
+    return { status, data, isSuccess: true, isLoading: false, isError: false };
   }
 
   if (status === 'error') {
-    return { status, error };
+    return { status, error, isSuccess: false, isLoading: false, isError: true };
+  }
+
+  if (status === 'loading') {
+    return { status, data: undefined, isSuccess: false, isLoading: true, isError: false };
   }
```

Comment:
we don't need all these props. `status` should be sufficient to infer the entire tagged union:

example:
```typescript
type ResponseProps = 'data' | 'error' | 'status';
export type Foo =
  | Pick<Extract<SomeQueryResult, { status: 'success' }>, ResponseProps>
  | Pick<Extract<SomeQueryResult, { status: 'error' }>, ResponseProps>
  | Pick<Extract<SomeQueryResult, { status: 'idle' }>, ResponseProps>
  | Pick<Extract<Result, { status: 'loading' }>, ResponseProps>;

declare const foo: Foo

if (foo.status === 'success') { 
// correct types
}
.... 
```

Reply:
in addition, it seems every consumer is casting the return type as `unknown` then `UseQueryResult` 

so the fix here should prevent that too